src/OrchardCore/OrchardCore.Data.YesSql/OrchardCoreBuilderExtensions.cs

```diff
@@ -161,5 +159,44 @@ public static OrchardCoreBuilder AddDataAccess(this OrchardCoreBuilder builder)
 
             return builder;
         }
+
+        private static IConfiguration GetStoreConfiguration(IServiceProvider sp)
+        {
+            var yesSqlOptions = sp.GetService<IOptions<YesSqlOptions>>().Value;
+
+            IConfiguration storeConfiguration = new YesSql.Configuration
+            {
+                CommandsPageSize = yesSqlOptions.CommandsPageSize,
+                QueryGatingEnabled = yesSqlOptions.QueryGatingEnabled,
+                ContentSerializer = new PoolingJsonContentSerializer(sp.GetService<ArrayPool<char>>()),
+            };
+
+            if (yesSqlOptions.IdGenerator != null)
+            {
+                storeConfiguration.IdGenerator = yesSqlOptions.IdGenerator;
```

Comment:
```csharp
 storeConfiguration.IdGenerator ??= yesSqlOptions.IdGenerator;
```

Reply:
YesSql.Configuration ctor already defines the IdGenerator so this will never be NULL. Though, if the intent is to replace the IdGenerator by a custom implementation then the code is fine. It should not use the ??= operator.